documentation/src/main/asciidoc/user_guide/integrations_hibernate_cache.adoc

```diff
@@ -0,0 +1,506 @@
+   value= "LIRS"/>
+<property name="hibernate.cache.infinispan._war_or_ear_name_#_unit_name_.com.acme.Person.expiration.lifespan"
+   value= "65000"/>
+----
+
+
+==== Cache Strategies
+
+Infinispan cache provider supports all Hibernate cache strategies:
+`transactional`, `read-write`, `nonstrict-read-write` and `read-only`.
+
+However, when running in a cluster configuration combinations are limited:
+
+* _non-transactional invalidation_ caches are supported with `read-write` strategy.
+The actual setting of cache concurrency mode (`read-write` vs. `transactional`) is not honored, the appropriate strategy is selected based on the cache configuration (_non-transactional_ vs. _transactional_).
+* `read-write` mode is supported on _non-transactional distributed/replicated_ caches, however, eviction should not be used in this configuration.
+Use of eviction can lead to consistency issues.
+Expiration (with reasonably long max-idle times) can be used.
+* `nonstrict-read-write` mode is supported on _non-transactional distributed/replicated_ caches, but the eviction should be turned off as well.
+In addition to that, the entities must use versioning.
+This mode mildly relaxes the consistency - between DB commit and end of transaction commit a stale read (see <<caching-provider-infinispan-stale-read-example,example>>) may occur in another transaction.
+However this strategy uses less RPCs and can be more performant than the other ones.
+* `read-only` mode is supported on both _transactional_ and _non-transactional_ _invalidation_ caches and _non-transactional distributed/replicated_ caches, but use of this mode currently does not bring any performance gains.
+
+The available combinations are summarized in table below:
+
+[[caching-provider-infinispan-compatibility-table]]
+.Cache concurrency strategy/cache mode compatibility table
+[options="header"]
+|===
+|Concurrency strategy|Cache transactions|Cache mode             |Eviction
+|transactional       |transactional     |invalidation           |yes
+|read-write          |non-transactional |invalidation           |yes
+|read-write          |non-transactional |distributed/replicated |no
+|nonstrict-read-write|non-transactional |distributed/replicated |no
+|===
+
+Changing caches to behave different to the default behaviour explained in previous section is explained in the <<_configuration_properties, configuration properties>> section.
+
+[[caching-provider-infinispan-stale-read-example]]
+.Stale read with `nonstrict-read-write` strategy
+====
+[source, indent=0]
+----
+A=0 (non-cached), B=0 (cached in 2LC)
+TX1: write A = 1, write B = 1
+TX1: start commit
+TX1: commit A, B in DB
+TX2: read A = 1 (from DB), read B = 0 (from 2LC) // breaks transactional atomicity
+TX1: update A, B in 2LC
+TX1: end commit
+Tx3: read A = 1, B = 1 // reads after TX1 commit completes are consistent again
+----
+====
+
+==== Remote Infinispan Caching
```

Comment:
This whole chapter is hypothetical and should be marked as such, otherwise users might think that they could already use the remote caching.

Reply:
True, I'll just remove it.